lib/opinionated_http/client.rb

```diff
@@ -49,22 +49,33 @@ def get(action:, path: "/#{action}", parameters: nil)
       path = "/#{path}" unless path.start_with?("/")
       path = "#{path}?#{URI.encode_www_form(parameters)}" if parameters
 
-      request  = Net::HTTP::Get.new(path)
+      request  = generic_request(path: path, verb: 'Get')
       response = request_with_retry(action: action, path: path, request: request)
 
       response.body
     end
 
-    def post(action:, path: "/#{action}", parameters: nil)
+    def post(action:, path: "/#{action}", headers: nil, body: nil, form_data: nil, auth: nil)
       path    = "/#{path}" unless path.start_with?("/")
-      request = Net::HTTP::Post.new(path)
-      request.set_form_data(parameters) if parameters
+      request = generic_request(path: path, verb: 'Post', headers: headers, body: body, form_data: form_data, auth: auth)
 
       response = request_with_retry(action: action, path: path, request: request)
 
       response.body
     end
 
+    # auth expects hash of the form {'username' =>, 'password' =>}
+    def generic_request(path:, verb:, headers: nil, body: nil, form_data: nil, auth: nil)
```

Comment:
Instead of using `:auth`, could we supply `:username` and `:password` directly as arguments?

Reply:
certainly